### PR TITLE
jni: expose memtable_whole_key_filtering option

### PR DESCRIPTION
**Pull request by javeme**
refer to: https://github.com/facebook/rocksdb/wiki/Prefix-Seek#configure-prefix-bloom-filter

### CODE REVIEW

**Review by alanpaxton — COMMENTED** [review]
Please could you change the documentation comment to say "Return boolean indicating whether whole key bloom filter is enabled in memtable". There's not really a mode as such, and it's best when  adding to the JNI to directly echo the C++ interface documentation, for clarity.

File: java/src/main/java/org/rocksdb/AdvancedMutableColumnFamilyOptionsInterface.java [hunk 1]
```diff
@@ -80,6 +80,25 @@ T setMemtablePrefixBloomSizeRatio(
...
   double memtablePrefixBloomSizeRatio();
 
+  /**
+   * Enable whole key bloom filter in memtable. Note this will only take effect
+   * if memtable_prefix_bloom_size_ratio is not 0. Enabling whole key filtering
+   * can potentially reduce CPU usage for point-look-ups.
+   *
+   * Default: false (disable)
+   *
+   * @param memtableWholeKeyFiltering the memtable whole key filtering mode
```
Comment (alanpaxton) [inline java/src/main/java/org/rocksdb/AdvancedMutableColumnFamilyOptionsInterface.java:90]: I think it's better to put "true iff whole key bloom filter is enabled in memtable" - it's not really a mode.

File: java/src/main/java/org/rocksdb/AdvancedMutableColumnFamilyOptionsInterface.java [hunk 2]
```diff
@@ -80,6 +80,25 @@ T setMemtablePrefixBloomSizeRatio(
...
+   * @return the reference to the current options.
+   */
+  T setMemtableWholeKeyFiltering(boolean memtableWholeKeyFiltering);
+
+  /**
+   * Returns whether a memtable whole key filtering mode will be used
```
Comment (alanpaxton) [inline java/src/main/java/org/rocksdb/AdvancedMutableColumnFamilyOptionsInterface.java:96]: Same thing about comments "Returns whether whole key bloom filter is enabled in memtable"

**Review by alanpaxton — COMMENTED** [review]
Nice little change. Just a couple of changes I've suggested to the text of method documentation, to keep them as consistent as possible with the C++ documentation.

**Comment by javeme** [comment]
@alanpaxton Thanks for your review. The documentation issue has been resolved now.

**Review by ajkr — APPROVED** [review]
LGTM

**Pull request closed without merging**